### DISCUSSION TITLE
Each SyncVar OnChange Not Invoked Until All SyncVars have been synced that Tick

### DISCUSSION THREAD

**Discussion by Katerlad** [FirstGearGames/FishNet#467] [Feature Request]
Request to revise  SyncVars OnChange behavior to not be invoked until all SyncVars have synced from the same Tick.

This was a request added back in the old request system in discord that was approved, and I have seen you mention it as something that you may like to change/add a few times here and there in people questions etc. 

So making an official thread about it here on Github, now that SyncVars are a hot topic with V4 maybe this could be another positive change.

**WHY?** 

Currently if 2 SyncVars were updated on the same callstack/tick when the OnChange callbacks fire on the clients, if the the first triggered callback depended on the 2nd Syncvar it would be using out of date info. 

```csharp
[SyncVar(OnChange = nameof(HandleFooChange)]
private int foo = 0;

[SynvVar(OnChange = nameof(HandleBarChange)]
private int bar = 0;

[Server]
public void ChangeFooBar()
{
    foo = 1;
    bar = 1;
}

//Lets say this Callback invoked first on the client
public void HandleFooChange(int oldValue, int newValue, bool asServer)
{
    //foo + bar = 1 because bar has not changed yet.
}

//Lets say this Callback invoked second on the client
public void HandleBarChange(int oldValue, int newValue, bool asServer)
{
    //foo + bar = 2 because both foo and bar values have finished syncing.
}
```

**Comment by FirstGearGames** [comment]
Bump

**Comment by FirstGearGames** [comment]
Not really possible without a huge rework and a lot of complications. Adding 'on all sync types complete' event instead.
